SDK/include/NDK/Widgets/CheckboxWidget.inl

```diff
@@ -0,0 +1,96 @@
+﻿// Copyright (C) 2017 Samy Bensaid
+// This file is part of the "Nazara Development Kit"
+// For conditions of distribution and use, see copyright notice in Prerequesites.hpp
+
+namespace Ndk
+{
+
+	inline void CheckboxWidget::EnableAdaptativeMargin(bool enable)
+	{
+		m_adaptativeMargin = enable;
+		Layout();
+	}
+
+	inline void CheckboxWidget::EnableCheckbox(bool enable)
+	{
+		m_checkboxEnabled = enable;
+		UpdateCheckbox();
+	}
+
+	inline void CheckboxWidget::EnableTristate(bool enable)
+	{
+		m_tristateEnabled = enable;
+
+		if (GetState() == CheckboxState_Tristate)
+			SetState(CheckboxState_Unchecked);
```

Comment:
It makes no sense to reset the state to Unchecked when enabling a Tristate state (EnableTristate(true) could be called on a widget already using tri-states)